geomstats/learning/em_expectation_maximization.py

```diff
@@ -0,0 +1,265 @@
+                         'contain elements that are not numbers')
+
+        num_normalized_pdf = gs.einsum('j,...j->...j',
+                                       self.mixture_coefficients,
+                                       probability_distribution_function)
+        valid_pdf_condition = gs.amin(gs.sum(num_normalized_pdf, -1))
+
+        if valid_pdf_condition <= PDF_TOL:
+
+            num_normalized_pdf[gs.sum(num_normalized_pdf, -1) <= PDF_TOL] = 1
+
+        sum_pdf = gs.sum(num_normalized_pdf, -1)
+        posterior_probabilities =\
+            gs.einsum('...i,...->...i', num_normalized_pdf, 1 / sum_pdf)
+
+        if (gs.mean(posterior_probabilities) !=
+                gs.mean(posterior_probabilities)):
+
+            raise NameError('EXPECTATION : posterior probabilities ' +
+                            'contain elements that are not numbers')
+
+        if 1 - SUM_CHECK_PDF >= gs.mean(gs.sum(
+                posterior_probabilities, 1)) >= 1 + SUM_CHECK_PDF:
+
+            raise NameError('EXPECTATION : posterior probabilities ' +
+                            'do not sum to 1')
+
+        return posterior_probabilities
+
+    def _maximization(self,
+                      data,
+                      posterior_probabilities,
+                      lr_means,
+                      conv_factor_mean,
+                      max_iter=DEFAULT_MAX_ITER):
+        """Update function for the means and variances."""
+        self.update_posterior_probabilities(posterior_probabilities)
+
+        if(gs.mean(self.mixture_coefficients)
+                != gs.mean(self.mixture_coefficients)):
+            raise NameError('UPDATE : mixture coefficients ' +
+                            'contain elements that are not numbers')
+
+        self.update_means(data,
+                          posterior_probabilities,
+                          lr_means=lr_means,
+                          tau_means=conv_factor_mean,
+                          max_iter=max_iter)
+
+        if self.means.mean() != self.means.mean():
+            raise NameError('UPDATE : means contain' +
+                            'not a number elements')
+
+        self.update_variances(data, posterior_probabilities)
+
+        if self.variances.mean() != self.variances.mean():
+            raise NameError('UPDATE : variances contain' +
+                            'not a number elements')
+
+    def fit(self,
+            data,
+            max_iter=DEFAULT_MAX_ITER,
+            lr_mean=DEFAULT_LR,
+            conv_factor_mean=DEFAULT_CONV_FACTOR):
+        """Fit a Gaussian mixture model (GMM) given the data.
+
+        Alternates between Expectation and Maximisation steps
+        for some number of iterations.
+
+        Parameters
+        ----------
+        data : array-like, shape=[n_samples, n_features]
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
```

Comment:
No blank lines between parameter descriptions
Missing periods at the end of parameter descriptions

see contributing for docstring standard in geomstats (follows np's conventions)